vassal-app/src/main/java/VASSAL/build/module/Console.java

```diff
@@ -0,0 +1,235 @@
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, copies are available
+ * at http://www.opensource.org.
+ */
+package VASSAL.build.module;
+
+import VASSAL.Info;
+import VASSAL.build.GameModule;
+import VASSAL.build.module.properties.MutableProperty;
+import VASSAL.command.Logger;
+import VASSAL.tools.BugUtils;
+import VASSAL.tools.SequenceEncoder;
+import org.slf4j.LoggerFactory;
+
+import java.awt.Desktop;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+/**
+ * Expandable "Console" to allow entering commands into the Chatter.
+ */
+public class Console {
+  SequenceEncoder.Decoder decode;
+  String commandLine;
+
+  private static final org.slf4j.Logger log =
+    LoggerFactory.getLogger(Console.class);
+
+  private void show(String s) {
+    GameModule.getGameModule().warn(s);
+  }
+
+
+  private boolean matches(String s1, String s2) {
+    return matches(s1, s2, 2);
+  }
+
+  private boolean matches(String s1, String s2, int min) {
+    if (s2.isEmpty() || (s2.length() > s1.length()) || ((s2.length() < min) && (s1.length() > s2.length()))) {
+      return false;
+    }
+    if (s2.length() < min) {
+      return s1.toLowerCase().equals(s2.toLowerCase());
+    }
+    return s1.toLowerCase().substring(0, s2.length()).equals(s2.toLowerCase());
+  }
+
+
+  private boolean doErrorLog() {
+    final String option = decode.nextToken("");
+    if (matches("show", option)) { //NON-NLS
+      final String errorLog = BugUtils.getErrorLog();
+      final String delims2 = "[\n]+";
+      final String[] lines = errorLog.split(delims2);
+
+      if (lines.length > 0) {
+        final int end = lines.length - 1;
+        final int linesToShow = decode.nextInt(0);
+        final int start = (linesToShow <= 0) ? 0 : Math.max(0, end - linesToShow + 1);
+
+        for (int line = start; line <= end; line++) {
+          show(lines[line]);
+        }
+      }
+    }
+    else if (matches("write", option)) { //NON-NLS
+      final int where = commandLine.toLowerCase().indexOf("write"); //NON-NLS
+      if ((where > 0) && commandLine.length() > where + 6) {
+        log.info(commandLine.substring(where + 6));
+      }
+    }
+    else if (matches("folder", option)) { //NON-NLS
+      final Desktop desktop = Desktop.getDesktop();
+      try {
+        final File dirToOpen = Info.getConfDir();
+        desktop.open(dirToOpen);
+      }
+      catch (IOException e) {
+        System.out.println("File Not Found"); //NON-NLS
+      }
+      catch (IllegalArgumentException iae) {
+        System.out.println("Illegal argument"); //NON-NLS
+      }
```

Comment:
You've got this twice. Extract to a function?

Reply:
done